File: src-tauri/src/models/quiz.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum QuestionType {
    Single,
    Multiple,
    Short,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QuizQuestion {
    pub id: String,
    pub question_type: QuestionType,
    pub question: String,
    pub options: Vec<String>,
    pub answer: String,
    pub explanation: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QuizStreamParams {
    pub path: String,
    pub types: Vec<QuestionType>,
    pub count: u32,
    pub difficulty: String,
    pub lang: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QuizResult {
    pub total: u32,
    pub questions: Vec<QuizQuestion>,
}

File: src/types/quiz.ts
export type QuestionType = 'single' | 'multiple' | 'short'
export type QuizDifficulty = 'easy' | 'medium' | 'hard'
export type QuizLanguage = 'zh' | 'en' | 'ja' | 'ko'

export interface QuizQuestion {
  id: string
  question_type: QuestionType
  question: string
  options: string[]
  answer: string
  explanation: string
}

export interface QuizStreamParams {
  path: string
  types: QuestionType[]
  count: number
  difficulty: QuizDifficulty
  lang: QuizLanguage
}

export type QuizEvent =
  | { event: 'chunk'; data: QuizQuestion }
  | { event: 'done'; data: { total: number } }
  | { event: 'error'; data: { message: string } }
